### DISCUSSION TITLE
umi-max创建的项目Procomponent组件没有代码提示

### DISCUSSION THREAD

**Discussion by loveCHN** [umijs/umi#12288] [Q&A]
ant-design/pro-components正常输入无代码提示
![82487c2aaef12c0ec78fd50514a2023](https://github.com/umijs/umi/assets/98575263/c01f909f-1e65-4692-8df4-bc667b2ecc31)
在顶部引入有正常提示
![1713239766126](https://github.com/umijs/umi/assets/98575263/5ea05879-dce8-421c-a3eb-460536859a08)
antd是正常的
![1713239860186](https://github.com/umijs/umi/assets/98575263/b695ff17-f299-4000-9455-f04606e98bf9)
以下是我的package.json
![1713239917507](https://github.com/umijs/umi/assets/98575263/c5d760aa-2be8-4ca0-b6cb-f858f86e5abe)

**Comment by fz6m** [comment]
你用的是什么包管理器啊，有最小复现吗。

如只有在你的电脑上可以复现的话，可以提供一个视频帮助排查问题。

**Reply by loveCHN** [reply]
我用的是pnpm 使用  pnpm dlx create-umi@latest创建的项目  选择的是ant design pro的模板 生成出来之后就是这样了  只有Procomponent没有代码提示 需要在顶部手动引入

**Reply by fz6m** [reply]
是什么 pnpm 版本，使用 pnpm v8 / v9 + vscode 最新版本也有这个问题吗，。

**Reply by loveCHN** [reply]
vscode 版本：1.88.1
pnpm：8.15.5

**Reply by loveCHN** [reply]
有的时候有提示有的时候没有提示  大部分时间没有提示  小部分时候又有提示

**Reply by fz6m** [reply]
1. 确保 vscode 一次只打开一个项目，并且这个项目的根目录就是 vscode 当前窗口的根目录，也就是 tsconfig.json 就在第一层。

2. 观察 vscode 右下角正在使用的 TS 版本，点击它，选择使用项目内的 TS 版本。

3. 确保排除其他 vscode 插件可能的影响，比如有很多插件冲突了（自己可以 review 下插件列表）。

4. 如果开发时类型提示失效了，没有下拉列表，提前配一下触发智能提示的快捷键，vscode shortcut command id 是 `editor.action.triggerSuggest` ，强开一下下拉列表看看。

5. 如果还不行，就 ctrl + shift + p 输入 restart ts 重开 TS 引擎（可能是机器性能不好，建议换更好的开发机，因为 umi 生成了太多临时文件，加载的类型比较多，在性能一般的机子上出现这种情况也是可能的）。
